src/result.js

```diff
@@ -1305,14 +1305,14 @@ var Result = CreateClass({
 
                         var supplementalDamageInfo = [];
                         const supplementalInfo = supplemental.collectSkillInfo(skilldata.supplementalDamageArray, {remainHP: m.data[key].remainHP});
-                        if ("keys" in supplementalInfo && "values" in supplementalInfo && "total" in supplementalInfo) {
+                        if ("keys" in supplementalInfo && "values" in supplementalInfo && "total" in supplementalInfo && supplementalInfo.total > 0) {
                             supplementalDamageInfo.push(
                                 <table key={key + "-supplementalDamageTable"} className="table table-bordered" style={{"marginBottom": "0px"}} >
                                     <thead>
                                         <tr>
                                             <th className="bg-success" style={{"fontSize": "10pt"}}>{intl.translate("与ダメージ上昇効果のソース", locale)}</th>
                                             {supplementalInfo.keys.map( function (v, ind) {
-                                                return <th key={ind} className="bg-success" style={{"fontSize": "10pt"}}>{intl.translate(v, locale)}</th>
+                                                return <th key={ind} className="bg-success" style={{"fontSize": "10pt"}}>{intl.translate(v[0], locale) + intl.translate(("supplemental_"+v[1]), locale)}</th>
```

Comment:
- [ ] <th key should be unique?
  - https://reactjs.org/docs/lists-and-keys.html#keys
- [ ] arrow function can make omit "function" "return"

Reply:
>  \<th key should be unique?

it's unique, is it not?

>  arrow function can make omit "function" "return"
 
?

Reply:
> arrow function can make omit "function" "return"

I should type "arrow-function"
it used when write a function in a expression.
```js
{supplementalInfo.keys.map((v, ind) => <th>...</th>)}
```
used in map(), .filter() usually.

> it's unique, is it not?

it's ok. seem that was safe case to use index as keys.
I thought "key" of supplementalInfo could the unique.

> Keys Must Only Be Unique Among Siblings
> Keys used within arrays should be unique among their siblings. However they don’t need to be globally unique. We can use the same keys when we produce two different arrays:

index is ok here. now our contents is static.
Document said that may break unique when list item change dynamically.

> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state.

https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318
> When all of them are met, you may safely use the index as a key.

- [x] 1. the list and items are static–they are not computed and do not change;
- [ ] 2. the items in the list have no ids;
  - or "key" of supplementalInfo can be the key for React <th>
    it's unique keys of supplementalDamageArray
- [x] 3. the list is never reordered or filtered.
  - it filtered hp-based, but not dynamic filtering ... it's re-calc and generate with new indexes.